src/traces/isosurface/convert.js

```diff
@@ -0,0 +1,480 @@
+            (data.meshalgo === 'SurfaceNets') ?
+                createIsosurface.surfaceNets(dims, fXYZs) :
+            (data.meshalgo === 'MarchingTetrahedra') ?
+                createIsosurface.marchingTetrahedra(dims, fXYZs) :
+                createIsosurface.marchingCubes(dims, fXYZs); // default: MarchingCube
+
+        var q, len;
+
+        var positions = isosurfaceMesh.positions;
+        len = positions.length;
+
+        var axis;
+
+        var starts = [
+            Xs[0],
+            Ys[0],
+            Zs[0]
+        ];
+        var ends = [
+            Xs[Xs.length - 1],
+            Ys[Ys.length - 1],
+            Zs[Zs.length - 1]
+        ];
+
+        // map pixel coordinates (0..n) to (real) world coordinates
+        for(axis = 0; axis < 3; axis++) {
+            var start = starts[axis];
+            var end = ends[axis];
+            for(q = 0; q < len; q++) {
+                positions[q][axis] = start + (end - start) * positions[q][axis] / (dims[axis] - 1);
+            }
+        }
+
+        // handle non-uniform 3D space
+        for(axis = 0; axis < 3; axis++) {
+            var xyz = (axis === 0) ? data.x : (axis === 1) ? data.y : data.z;
+
+            for(q = 0; q < len; q++) {
+                var here = positions[q][axis];
+
+                var minDist = Infinity;
+                var first = 0;
+                var second = 0;
+
+                var dist;
+                for(i = 0; i < dims[axis]; i++) {
+                    dist = Math.abs(here - xyz[i]);
+
+                    if(dist < minDist) {
+                        minDist = dist;
+                        second = first;
+                        first = i;
+                    }
+                }
+
+                if(dist > 0 && first !== second) {
+
+                    var d1 = -(here - xyz[first]);
+                    var d2 = (here - xyz[second]);
+
+                    positions[q][axis] = (xyz[first] * d2 + xyz[second] * d1) / (d1 + d2);
+                }
+            }
+        }
+
+        // record positions & colors of iso surface
+        for(q = 0; q < len; q++) {
+            allXs.push(positions[q][0]);
+            allYs.push(positions[q][1]);
+            allZs.push(positions[q][2]);
+
+            allCs.push(intensity);
+        }
+
+        // record cells of iso surface
+        var cells = isosurfaceMesh.cells;
+        len = cells.length;
+        for(q = 0; q < len; q++) {
+            data.i.push(cells[q][0] + num_vertices);
+            data.j.push(cells[q][1] + num_vertices);
+            data.k.push(cells[q][2] + num_vertices);
+        }
+
+        num_vertices += positions.length;
+    }
+
+    data.x = allXs;
+    data.y = allYs;
+    data.z = allZs;
+    data.intensity = allCs;
+
+    return data;
+}
+
+function createIsosurfaceTrace(scene, data) {
+
+    var gl = scene.glplot.gl;
+
+    var mesh = createMesh({gl: gl});
```

Comment:
Reusing `gl-mesh3d` is a very nice idea. One drawback I noticed though: hover labels don't appear on the input data coordinates, they appear on the (interpolated) mesh x/y/z - which I think is _ok_ for now, albeit not consistent with our other trace types. For example, surface hover always picks out data-coordinates (even when the surface coordinates are refined). In contrast, [streamtubes](https://rreusser.github.io/plotly-mock-viewer/#gl3d_streamtube-thin) hover picks the coordinates on the tubes themselves.